### ISSUE TITLE
RELEASES file contains duplicate rows

### ISSUE THREAD

**Issue by plinehan** [Squirrel/Squirrel.Windows#578]
On a certain hardware setup (Windows inside VirtualBox on a Mac) Squirrel is always generating a RELEASES file with duplicate lines, e.g.

== BEGIN ==
FD9BEF752A8E3CFA1E2A11A1FAB703F530443A9D Foo-4.4.14-full.nupkg 48770679
FD9BEF752A8E3CFA1E2A11A1FAB703F530443A9D Foo-4.4.14-full.nupkg 48770679
== END ==

This causes a problem when clients attempt to upgrade and they blow up here because there isn't a "single" result:

http://bit.ly/1RPDGWH

Any idea where the duplicate lines could be coming from?  I'm building in a VirtualBox shared folder, which is to say the files are really on the Mac but are mounted in Windows as some kind of networked drive.  The code that writes the RELEASES file looks reasonable and atomic, so I'm at a bit of a loss.

**Comment by paulcbetts** [comment]
That's super weird, but I guess if you're desperate you could always pipe it through `uniq`. RELEASES was intentionally designed to be super simple and editable

**Comment by plinehan** [comment]
Yeah I'm just fixing it up myself for now.  Not a big deal but thought you might have some insight as to the cause.

Would you be averse to changing the code to use "FirstOrDefault" instead of "SingleOrDefault"?

**Comment by paulcbetts** [comment]
I think in this case we're hiding the Real Bug then

**Comment by plinehan** [comment]
No, yeah, I feel ya.  To play devil's advocate, this code is reading from a file that is designed to be human -editable, and it should fail gracefully on any human input data it receives.  Like, in this case, the penalty for a slightly-malformed RELEASES file was [potentially] a hard crash of the app.  I have all Squirrel calls wrapped in a try/catch so my app was safe, but others might not be so lucky.

No need to respond in detail if you disagree; that's fine.  I just think it is worth considering.

I'll see if I can track down the root cause, though, of how the double-line got there in the first place.  Truly strange.  Possibly a poltergeist?

**Comment by plinehan** [comment]
And thanks for Squirrel!

**Comment by dataich** [comment]
Same for me 😢

**Comment by adamhathcock** [comment]
Odd.  I'm having the same issue when using VirtualBox on OS X.

**Comment by adamhathcock** [comment]
At least, I know the workaround now.  Thanks for this.
